etl/steps/data/garden/un/2024-02-14/sdgs_urbanization.py

```diff
@@ -0,0 +1,53 @@
+"""Load a meadow dataset and create a garden dataset."""
+
+from owid.catalog import Table
+
+from etl.data_helpers import geo
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    ds_meadow = paths.load_dataset("sdgs_urbanization")
+
+    # Read table from meadow dataset.
+    tb = ds_meadow["un_sdg"].reset_index()
+
+    metadata = tb.metadata
+
+    #
+    # Process data.
+    #
+    tb = geo.harmonize_countries(df=tb, countries_file=paths.country_mapping_path)
+
+    average_by_country = tb.groupby(["country", "year", "seriesdescription"])["value"].mean().reset_index()
+
+    average_by_country = average_by_country.dropna(subset=["value"])
```

Comment:
Nit - grouping by categorical variables in pandas by default creates a group even for **non-observed categories**. That's why you end up with so many NaNs. It is ok since the dataset is small, but the dataframe can explode with bigger datasets.

If you add `groupby(..., observed=True)`, it'll only keep non-null values. (I don't know why isn't it default)